### PR TITLE
FlightTasks: switch to uORB::Subscription

### PR DESCRIPTION
**Pull request by dagar**
Switching to uORB::Subscription from uORB::SubscriptionPollable is slightly faster and safer. There are no dynamic allocations, so we can get rid of the SubscriptionArray. The minor downside in this case is that there isn't a helper to update all of them in a single pass, so each needs to be updated in updateInitialize().

This is working towards removing SubscriptionPollable entirely. It's a wrapper around the orb C api.

### CODE REVIEW

**Comment by dagar** [comment]
This also saves about 1.1 KB of flash on fmu-v2.

File: src/lib/FlightTasks/FlightTasks.hpp [hunk 1]
```diff
@@ -55,7 +54,6 @@
 enum class FlightTaskError : int {
 	NoError = 0,
 	InvalidTask = -1,
-	SubscriptionFailed = -2,
 	ActivationFailed = -3
```
Comment (bkueng) [inline src/lib/FlightTasks/FlightTasks.hpp]:
```suggestion
	ActivationFailed = -2
```

Comment (dagar) [reply src/lib/FlightTasks/FlightTasks.hpp]: Fixed.

**Comment by dagar** [comment]
@PX4/testflights could you test this across all the normal multicopter modes?

**Comment by jorge789** [comment]
### Tested on PixRacer V4:
**Flight Card 1**

**Modes Tested:**
Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.

**Procedure:**
Arm in position mode, check take off behavior (Vehicle should stay on the ground as long as the throttle stick is at the middle or below, and start to ascend smoothly as long as throttle stick as above the middle.), the land and disarm.
Same procedure was repeated with Altitude and Stabilized mode.

**Notes:**
No issues were noted, good flight in general.

**Flight Card 2**

**Modes Tested**
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**
Arm and Take off in mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint see landing behavior.

**Note:**
No issues were noted, good flight in general.

**Flight Card 3**

**Modes Tested**
Position Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint see landing behavior.

**Note:**
No issues were noted, good flight in general.

**Flight Card 4**
Tested Fail safe:

**Procedure**
The radio control turned off and the vehicle returned and landed correctly.
Tested good.

**Logs:**
Flight card 1:  https://review.px4.io/plot_app?log=b42f99c1-730a-4ae1-9caa-660ee001055c
Flight card 1:  https://review.px4.io/plot_app?log=c31bed71-ac08-40e0-8f65-1b57dd19bbf2

Flight card 2: https://review.px4.io/plot_app?log=54654f33-f74d-4e12-89d7-7d3d08f684fc
Flight card 2: https://review.px4.io/plot_app?log=5c1225d9-16ff-4117-a241-6b23da9a0e7b

Flight card 3: https://review.px4.io/plot_app?log=ad324476-f95c-44ba-bf92-6c03ec8df39b
Flight card 3: https://review.px4.io/plot_app?log=5e623620-2d15-4603-95fd-586e109814e0

Flight card 4: https://review.px4.io/plot_app?log=baa4ab5b-e500-4dac-930e-19154ae7e240
Flight card 4: https://review.px4.io/plot_app?log=f095db29-19cf-4acb-b875-f1db632fd3bd

**Comment by dannyfpv** [comment]
Tested on Pixhawk4 V5:
Flight Card 1

Modes Tested:
Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.

Procedure:
Arm in position mode, check take off behavior (Vehicle should stay on the ground as long as the throttle stick is at the middle or below, and start to ascend smoothly as long as throttle stick as above the middle.), the land and disarm.
Same procedure was repeated with Altitude and Stabilized mode.

Notes:
No issues were noted, good flight in general.

Flight Card 2

Modes Tested
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

Procedure
Arm and Take off in mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint see landing behavior.

Note:
No issues were noted, good flight in general.

Flight Card 3

Modes Tested
Position Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

Procedure
Arm and Take off in position mode, after flying for approximately one minute, switched to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint see landing behavior.

Note:
No issues were noted, good flight in general.

Flight Card 4
Tested Fail safe:

Procedure
The radio control turned off and the vehicle returned and landed correctly.
Tested good.

Logs:
Flight card 1:  
https://review.px4.io/plot_app?log=84421f12-0c88-4582-b244-885c6b8d01c7
Flight card 2: 
https://review.px4.io/plot_app?log=68870e63-859a-47e1-b8d0-2421256e35d1
Flight card 3: 
https://review.px4.io/plot_app?log=d0c0c256-e0c1-447b-8a44-76675ffccc40
Flight card 4: 
https://review.px4.io/plot_app?log=d66ebf2c-8f21-46e2-a1db-9bfbbcdfd653

**Comment by dagar** [comment]
Thanks everyone.

**Pull request merged**